### PR TITLE
Add popup:set_position(position, relative)

### PR DESCRIPTION
**Pull request by fhill2**
supports popup:set_position(...) when unmounted or mounted

Also supports:
position (required. same as popup_options.position)
relative (optional, same as popup_options.relative, unchanged if not provided)

Let me know what you would change :D

Also, I think `set_size()` popup method needs this change:
```lua
if self.border.win_config then
  self.border:resize()
end
```
Because when using this in config to initialize popup():
```
border = {
style = "none",
}
```
makes `set_size()` error. `attempt to index field 'win_config' (a nil value)`

### CODE REVIEW

File: lua/nui/popup/border.lua [hunk 1]
```diff
@@ -384,6 +384,34 @@ function Border:resize()
   end
 end
 
+function Border:reposition()
```
Comment (fhill2) [inline lua/nui/popup/border.lua:387]: realized I didn't make Border:reposition accept new relative value

Comment (MunifTanjim) [reply lua/nui/popup/border.lua:387]: Border's method should not be called directly. So this method is not supposed to take any arguments.
Border should reposition itself based on the configuration of Popup.

File: lua/nui/popup/init.lua [hunk 2]
```diff
@@ -304,6 +304,56 @@ function Popup:set_size(size)
   end
 end
 
+function Popup:set_position(position, relative)
+  local props = self.popup_props
+
+  local container_info = get_container_info(self)
+  props.position = calculate_window_position(position, props.size, container_info)
+
+  self.border:reposition()
```
Comment (MunifTanjim) [inline lua/nui/popup/init.lua:313]: This should go after the new `self.win_config` of the Popup is set. Border will need to use these values from Popup.

Comment (fhill2) [reply lua/nui/popup/init.lua:313]: Ok awesome, let me try again with your changes.

File: lua/nui/popup/border.lua [hunk 3]
```diff
@@ -391,25 +391,30 @@ function Border:reposition()
...
-    win_config.bufpos = self.win_config.bufpos
+  if self.popup.win_config.bufpos then
+    self.win_config.bufpos = self.popup.win_config.bufpos
+    self.win_config.win = nil
   end
 
-  if self.win_config.win then
-    win_config.win = self.win_config.win
+  if self.popup.win_config.win then
+    self.win_config = self.popup.win_config.win
```
Comment (fhill2) [inline lua/nui/popup/border.lua:402]: should be:
```lua
self.win_config.win = self.popup.win_config.win
```

Comment (MunifTanjim) [reply lua/nui/popup/border.lua:402]: Let me know when you're finished working on this 😃

Comment (fhill2) [reply lua/nui/popup/border.lua]: @MunifTanjim finished!

Comment (fhill2) [reply lua/nui/popup/border.lua]: Not sure why stylua CI is failing, I formatted all files with the .stylua config in the repo.

Comment (MunifTanjim) [reply lua/nui/popup/border.lua]: No worries. I'll squash the commits into a single commit and format again.

By the way, the commits don't have your right email address associated with your github account. Is that intentional? Or do you want to fix that? 🤔

Comment (fhill2) [reply lua/nui/popup/border.lua]: That's my bad, I switched OS and forgot to update it. Thank you for pointing that out. 
I've updated it, `2744926` is with my github. I'm guessing when you squash you can use the author/email from that?

Comment (MunifTanjim) [reply lua/nui/popup/border.lua]: > I'm guessing when you squash you can use the author/email from that?

Yep. Thanks! 😃

**Comment by MunifTanjim** [comment]
Hello @fhill2, I finally got the chance to test this. I've rebased the branch on main and pushed an extra commit that handles some corner-cases that I found during testing.

Let me know if that looks okay to you or if you want to change anything. And then we can merge it 😃

**Comment by fhill2** [comment]
@MunifTanjim Looks awesome! Let's merge!

**Review by MunifTanjim — APPROVED** [review]
